### PR TITLE
Warn user on libraries placeholder

### PR DESCRIPTION
**Pull request by iFrostizz**
As per https://github.com/runtimeverification/evm-semantics/issues/1901#issuecomment-1591385348
We are now asking the user to link the libraries in anticipation to a more long-term solution.

### CODE REVIEW

File: kevm-pyk/src/kevm_pyk/solc_to_k.py [hunk 1]
```diff
@@ -379,10 +379,17 @@ def production(self) -> KProduction:
...
+            raise AssertionError(
+                'Some library placeholders have been found. Please link the library(ies) first. Ref: https://docs.soliditylang.org/en/v0.8.20/using-the-compiler.html#library-linking'
+            )
         return KRule(
             KRewrite(KEVM.bin_runtime(KApply(self.klabel)), KEVM.parse_bytestack(stringToken('0x' + self.bytecode)))
         )
 
+    def has_unlinked(self) -> bool:
+        return any(self.bytecode[i] == '__' for i in range(0, len(self.bytecode), 2))
+
```
Comment (ehildenb) [inline kevm-pyk/src/kevm_pyk/solc_to_k.py:392]: It seems a simpler solution (and probably faster):

```suggestion
    def has_unlinked(self) -> bool:
        return 0 <= self.bytecode.find('__')

```

Comment (ehildenb) [reply kevm-pyk/src/kevm_pyk/solc_to_k.py:392]: image

File: kevm-pyk/src/kevm_pyk/solc_to_k.py [hunk 2]
```diff
@@ -379,10 +379,17 @@ def production(self) -> KProduction:
 
     @property
     def macro_bin_runtime(self) -> KRule:
+        if self.has_unlinked():
+            raise AssertionError(
```
Comment (ehildenb) [inline kevm-pyk/src/kevm_pyk/solc_to_k.py:383]:
```suggestion
            raise ValueError(
```

File: kevm-pyk/src/kevm_pyk/solc_to_k.py [hunk 3]
```diff
@@ -379,10 +379,17 @@ def production(self) -> KProduction:
...
+                'Some library placeholders have been found. Please link the library(ies) first. Ref: https://docs.soliditylang.org/en/v0.8.20/using-the-compiler.html#library-linking'
```
Comment (ehildenb) [inline kevm-pyk/src/kevm_pyk/solc_to_k.py:384]:
```suggestion
                f'Some library placeholders have been found in contract {self.name}. Please link the library(ies) first. Ref: https://docs.soliditylang.org/en/v0.8.20/using-the-compiler.html#library-linking'
```

File: kevm-pyk/src/kevm_pyk/solc_to_k.py (hunk 1 above)
Comment (iFrostizz) [reply kevm-pyk/src/kevm_pyk/solc_to_k.py:392]: Nice thanks!

**Review by anvacaru — APPROVED** [review]

**Pull request merged**